test/functional/wallet_blank.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php.
+
+import os
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.address import (
+    ADDRESS_BCRT1_UNSPENDABLE,
+    ADDRESS_BCRT1_UNSPENDABLE_DESCRIPTOR,
+)
+from test_framework.key import ECKey
+from test_framework.util import (
+    assert_equal,
+)
+from test_framework.wallet_util import generate_wif_key
+
+
+class WalletBlankTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def test_createwallet(self):
+        self.log.info("Test that createwallet sets the blank flag when disableprivatekeys")
+        self.nodes[0].createwallet(wallet_name="disable_privs_w_blank", disable_private_keys=True)
+        wallet = self.nodes[0].get_wallet_rpc("disable_privs_w_blank")
+        assert_equal(wallet.getwalletinfo()["blank"], True)
+
+    def test_createwallet_override(self):
+        self.log.info("Test that the blank flag can be overridden")
+        self.nodes[0].createwallet(wallet_name="disable_privs_wo_blank", disable_private_keys=True, blank=False)
+        wallet = self.nodes[0].get_wallet_rpc("disable_privs_wo_blank")
+        assert_equal(wallet.getwalletinfo()["blank"], False)
+
+    def test_importaddress(self):
+        if self.options.descriptors:
+            return
+        self.log.info("Test that importaddress unsets the blank flag")
+        self.nodes[0].createwallet(wallet_name="iaddr", disable_private_keys=True)
+        wallet = self.nodes[0].get_wallet_rpc("iaddr")
+        assert_equal(wallet.getwalletinfo()["blank"], True)
+        wallet.importaddress(ADDRESS_BCRT1_UNSPENDABLE)
+        assert_equal(wallet.getwalletinfo()["blank"], False)
+
+    def test_importpubkey(self):
+        if self.options.descriptors:
+            return
+        self.log.info("Test that importpubkey unsets the blank flag")
+        self.nodes[0].createwallet(wallet_name="ipub", disable_private_keys=True)
+        wallet = self.nodes[0].get_wallet_rpc("ipub")
+        assert_equal(wallet.getwalletinfo()["blank"], True)
+
+        eckey = ECKey()
+        eckey.generate()
+
+        wallet.importpubkey(eckey.get_pubkey().get_bytes().hex())
+        assert_equal(wallet.getwalletinfo()["blank"], False)
+
```

Comment:
Once we import private keys, does the flag `private_keys_enabled` change its value to `true`? If so, shouldn't we also be checking for the flag `private_keys_enabled` is `true` now, here in the test?

Reply:
No, `WALLET_FLAG_DISABLE_PRIVATE_KEYS` is a permanent flag. It cannot be changed, and attempting to import private keys into a wallet with it set will result in an error and nothing imported. This test is for importing public keys, not private keys.

Reply:
Oh! Right, my bad, thanks for the clarification! I think I got confused with the following test which is `test_importprivkey`, but it doesn't have `private_keys_enabled` set to false either. I see what you're saying about the error trying to import private keys into a wallet with `WALLET_FLAG_DISABLE_PRIVATE_KEYS` on another test:
https://github.com/bitcoin/bitcoin/blob/3c1e75ef607000b265c15b47b32485cd95c1245d/test/functional/wallet_importdescriptors.py#L225-L230